concepts/05 Widgets/DataGrid/00 Getting Started with DataGrid/40 Edit and Validate Data.md

```diff
@@ -0,0 +1,187 @@
+                    <DxRequiredRule />
+                </DxColumn>
+                <DxColumn ...
+                    data-field="BirthDate">
+                    <DxRequiredRule />
+                </DxColumn>
+                <DxColumn ...
+                    data-field="HireDate">
+                    <DxRequiredRule />
+                </DxColumn>
+                <!-- ... -->
+                <DxColumn ...
+                    data-field="Country">
+                    <DxRequiredRule />
+                </DxColumn>
+                <DxEditing
+                    mode="popup"
+                    :allow-updating="true"
+                    :allow-adding="true"
+                    :allow-deleting="true"
+                />
+            </DxDataGrid>
+        </div>
+    </template>
+
+    <script>
+    import {
+        DxDataGrid,
+        DxColumn,
+        // ...
+        DxRequiredRule,
+        DxEditing
+    } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            DxColumn,
+            // ...
+            DxRequiredRule,
+            DxEditing
+        },
+        // ...
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import {
+        DataGrid,
+        Column,
+        // ...
+        RequiredRule,
+        Editing
+    } from 'devextreme-react/data-grid';
+
+    function App() {
+        return (
+            <div className="App">
+                <DataGrid ... >
+                    {/* ... */}
+                    <Column dataField="FullName">
+                        <RequiredRule />
+                    </Column>
+                    <Column dataField="Position">
+                        <RequiredRule />
+                    </Column>
+                    <Column ...
+                        dataField="BirthDate">
+                        <RequiredRule />
+                    </Column>
+                    <Column ...
+                        dataField="HireDate">
+                        <RequiredRule />
+                    </Column>
+                    {/* ... */}
+                    <Column ...
+                        dataField="Country">
+                        <RequiredRule />
+                    </Column>
+                    <Editing
+                        mode="popup"
+                        allowUpdating={true}
+                        allowDeleting={true}
+                        allowAdding={true}
+                    />
+                </DataGrid>
+            </div>
+        );
+    }
+
+    export default App;
+
+---
+
+Run the code and click the Edit button in any row. You will see a popup window that contains an edit form. Remove a value from the Full Name text box on this form. You should see the following validation error: *"First Name is required"*.
```

Comment:
>Remove a value from the Full Name text box on this form.

I would suggest to clarify that users have to use UI. Something like "Clear the 'Full Name' text box in(?) this form. You will see the..."

Reply:
As for "on the form" vs "in the form", it looks like it is "on" after all: https://www.learnersdictionary.com/qa/IN-or-ON-with-written-documents-page-blog-website-form-

```suggestion
Run the code and click the Edit button in any row. You will see a popup window that contains an edit form. Clear the Full Name text box on this form. You should see the following validation error: *"First Name is required"*.
```

Reply:
LGTM